C/Cpp_include/access_lock.hh

```diff
@@ -35,128 +35,135 @@ namespace litecore {
 #else
-            access(access&&) = delete;       // I cannot be moved
+            access(access &&) = delete;  // I cannot be moved
 #endif
 
-            Retained<R>& get()              {return _ref;}
-            operator R* ()                  {return _ref;}
-            R* operator -> ()               {return _ref;}
+            Retained<R> &get() { return _ref; }
+
+            operator R *() { return _ref; }
+
+            R *operator->() { return _ref; }
 
-        private:
+          private:
             friend access_lock;
-            access(MUTEX &mut, Retained<R> &ref, SENTRY& sentry):access(mut, ref) {
-                if (sentry) sentry(_ref);
+
+            access(MUTEX &mut, Retained<R> &ref, SENTRY &sentry) : access(mut, ref) {
+                if ( sentry ) sentry(_ref);
             }
+
             access_lock::LOCK_GUARD _lock;
-            Retained<R>&            _ref;
+            Retained<R>            &_ref;
         };
 
         // (specialization for const access when T is Retained<>)
         template <class R>
-        class access<const Retained<R>&> {
-        public:
-            access(MUTEX &mut, const R* ref):_lock(mut), _ref(ref) { }
-            access(const access&) =delete;  // I cannot be copied
+        class access<const Retained<R> &> {
+          public:
+            access(MUTEX &mut, const R *ref) : _lock(mut), _ref(ref) {}
+
+            access(const access &) = delete;  // I cannot be copied
 #if defined(_MSC_VER) && _MSC_VER < 1920
             // https://developercommunity.visualstudio.com/t/15935-guaranteed-copy-elision-failure/1398603
             // Deleting this constructor causes copy elision failure.  Microsoft won't fix prior to VS 2019.
-            access(access&&) { throw std::runtime_error("No moving!"); }
+            access(access &&) { throw std::runtime_error("No moving!"); }
 #else
-            access(access&&) = delete;       // I cannot be moved
+            access(access &&) = delete;  // I cannot be moved
 #endif
 
-            auto get()                      {return _ref;}
-            operator const R* ()            {return _ref;}
-            auto operator * ()              {return _ref;}
-            auto operator -> ()             {return _ref;}
+            auto get() { return _ref; }
+
+            operator const R *() { return _ref; }
+
+            auto operator*() { return _ref; }
 
-        private:
+            auto operator->() { return _ref; }
+
+          private:
             friend access_lock;
-            access(MUTEX &mut, const Retained<R>& ref, SENTRY& sentry):access(mut, ref.get()) {
-                if (sentry) sentry(ref);
+
+            access(MUTEX &mut, const Retained<R> &ref, SENTRY &sentry) : access(mut, ref.get()) {
+                if ( sentry ) sentry(ref);
             }
+
             access_lock::LOCK_GUARD _lock;
-            const R*                _ref;
+            const R                *_ref;
```

Comment:
Not a big fan of this format, as it's easy to glance down the types and miss that this is `R*` instead of just `R`

Reply:
There will probably be numerous examples of this throughout the code - where we have broken the rules, and Clang-Tidy would not be aware we have broken the rules. Each one either needs formatting manually, and disabling Clang-Tidy for that section of code, or needs accepting the regular rule.